server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -465,7 +469,7 @@ public static void registerRequestHandler(TransportService transportService, Sea
         );
         TransportActionProxy.registerProxyAction(transportService, QUERY_SCROLL_ACTION_NAME, true, ScrollQuerySearchResult::new);
 
-        transportService.registerRequestHandler(
+        registerRequestHandlerWithAPMMetrics(
```

Comment:
Hi @javanna , I'm a bit unsure as to what some of these phases/actions do, can you please check that I've instrumented the right ones here? Thanks!

Reply:
we may want to instrument FETCH_ID and CAN_MATCH as well, but before we do that, what is the purpose of this instrumentation? 

This traces how long it takes to execute the phase on each shard. That is very granular and probably useful. Another place to start, or possibly both are good, is to track how long each phase takes overall, from a coordinator perspective. What do you think?

Reply:
Thanks for the feedback! 
> We may want to instrument FETCH_ID and CAN_MATCH

Can do!

> Another place to start, or possibly both are good, is to track how long each phase takes overall, from a coordinator perspective.

Ah, good point.  I think I started here because, according to my notes, we discussed adding instrumentation here.   

Thinking about the current implementation a bit - the stats from each individual node will all collectively contribute to the same metric in Kibana, so one can still determine phase latencies/performance using instrumentation at this location.  

However, a there is a difference that comes to mind wrt collecting a similar metric at the coordinator.  When the phase is fanned out and collected, the overall phase time on the coordinator is that of the longest individual shard's phase, plus networking and coordination.  

If there exists an underlying issue whereby a single shard runs much longer than the others (for a given phase), the current metric will be able to expose that whereas an overall metric (from the coordinator's perspective) won't.  Theoretically, one could monitor the p99 (or equivalent) bucket in the histogram to try to detect these issues and use metric attributes to determine if the issue is common to the node/shard, or distributed across the cluster.  This could be useful information, at least in my mind 🙂 

I do think that the overall phase latency from the coordinator's perspective may be interesting as well in identifying overall trends and potentially answering other questions, but perhaps not enough on it's own.  At least that's my current thinking.  WDYT?